tools/project_manager/update_illos.php

```diff
@@ -0,0 +1,206 @@
+<?php
+// Replace an illustration file.
+
+$relPath = '../../pinc/';
+include_once($relPath.'base.inc');
+include_once($relPath.'theme.inc');
+include_once($relPath.'Project.inc');
+include_once($relPath.'misc.inc'); // get_upload_err_msg, get_enumerated_param, attr_safe, html_safe
+include_once($relPath.'project_states.inc'); // PROJ_NEW
+
+require_login();
+
+// (This script's functionality overlaps that of handle_bad_page.php.
+// They should perhaps be refactored.)
+
+$projectid = validate_projectID('projectid', @$_REQUEST['projectid']);
+
+$image     = @$_REQUEST['image'];
+// Note that using validate_page_image_filename() here would be inappropriate,
+// because it's specific to *page* images (which this isn't),
+// and so enforces a maximum filename length.
+if (!preg_match('/^\w[\w.-]*\.(png|jpg)$/', $image)) // see _check_file() in add_files.php
+{
+    // This should only happen if someone is URL-tweaking.
+    die(_("Parameter 'image' is not a valid illustration filename."));
+}
+
+$project = new Project($projectid);
+$operation = get_enumerated_param($_REQUEST, 'operation', 'replace', array('replace', 'delete'));
+
+$operation_image_str;
+if ($operation == 'replace') {
+    $operation_image_str = _('Replace Illustration');
+} else {
+    $operation_image_str = _('Delete Illustration');
+}
+
+output_header("$operation_image_str: {$project->nameofwork}");
+
+echo "<h1>" . html_safe($project->nameofwork) . "</h1>\n";
+echo "<h2>$operation_image_str: $image</h2>\n";
+
+if (!$project->can_be_managed_by_current_user)
+{
+    echo "<p>", _('You are not authorized to manage this project.'), "</p>\n";
+    return;
+}
+
+if ($operation == 'delete' && $project->state != PROJ_NEW)
+{
+    echo "<p>", _('You can only delete illustrations for a project in the new state.'), "</p>\n";
+    return;
+}
+
+$page_image_names = array();
```

Comment:
Consider moving lines 55-71 into their own function (`get_project_illustrations()`?). This not only helps bundle all of the functionality together but it helps document what this code is doing as well.

Bonus: put this function in the `Project` class as `get_illustrations()`.